hack/typelinter/typelinter.go

```diff
@@ -0,0 +1,107 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"go/types"
+	"os"
+	"regexp"
+
+	"golang.org/x/tools/go/packages"
+)
+
+type stringListVar map[string]struct{}
+
+func (sl stringListVar) String() string {
+	var keys []string
+	for k := range sl {
+		keys = append(keys, k)
+	}
+	return fmt.Sprintf("%v", keys)
+}
+
+func (sl *stringListVar) Set(value string) error {
+	if *sl == nil {
+		*sl = map[string]struct{}{}
+	}
+	(*sl)[value] = struct{}{}
+	return nil
+}
+
+var (
+	excludedFields stringListVar
+	typesWhitelist string
+
+	typesWhitelistRegexp *regexp.Regexp
+)
+
+func validStruct(structName string, s *types.Struct) bool {
```

Comment:
doc an example of what the structName looks like for `string`, `PodSpec`, and `time.Duration`